### ISSUE TITLE
Timers: HAL_TIM_PWM_Stop() disable timer even channel 5 or channel 6 are enabled

### ISSUE THREAD

**Issue by KamilDuljas** [STMicroelectronics/STM32CubeF7#106]
TIM_CCER_CCxE_MASK makro should be contain 6 channels, not  4. Missing `TIM_CCER_CC5E` and `TIM_CCER_CC6E`
![image](https://github.com/STMicroelectronics/STM32CubeF7/assets/87798948/6e41f934-fdbb-49d4-a24e-3cc532120eb9)

Example:
```
HAL_TIM_PWM_Start(&htim1,TIM_CHANNEL_1);
HAL_TIM_PWM_Start(&htim1,TIM_CHANNEL_2);
HAL_TIM_PWM_Start(&htim1,TIM_CHANNEL_5);

HAL_TIM_PWM_Stop(&htim1,TIM_CHANNEL_1);
// TIM_CR1_CEN is '1' (TIM_BDTR_MOE is '1' for  advanced timers)
HAL_TIM_PWM_Stop(&htim1,TIM_CHANNEL_2);
// TIM_CR1_CEN is '0' but CHANNEL 5 is still active!!!
HAL_TIM_PWM_Stop(&htim1,TIM_CHANNEL_5);
```

The issue related to macros TIM_CCER_CCxE_MASK  and:

```
/**
  * @brief  Disable the TIM peripheral.
  * @param  __HANDLE__ TIM handle
  * @retval None
  */
#define __HAL_TIM_DISABLE(__HANDLE__) \
  do { \
    if (((__HANDLE__)->Instance->CCER & TIM_CCER_CCxE_MASK) == 0UL) \
    { \
      if(((__HANDLE__)->Instance->CCER & TIM_CCER_CCxNE_MASK) == 0UL) \
      { \
        (__HANDLE__)->Instance->CR1 &= ~(TIM_CR1_CEN); \
      } \
    } \
  } while(0)

/**
  * @brief  Disable the TIM main Output.
  * @param  __HANDLE__ TIM handle
  * @retval None
  * @note The Main Output Enable of a timer instance is disabled only if all the CCx and CCxN channels have been
  *       disabled
  */
#define __HAL_TIM_MOE_DISABLE(__HANDLE__) \
  do { \
    if (((__HANDLE__)->Instance->CCER & TIM_CCER_CCxE_MASK) == 0UL) \
    { \
      if(((__HANDLE__)->Instance->CCER & TIM_CCER_CCxNE_MASK) == 0UL) \
      { \
        (__HANDLE__)->Instance->BDTR &= ~(TIM_BDTR_MOE); \
      } \
    } \
  } while(0)
```

Also please look closer `TIM_CCER_CCxNE_MASK`

**Comment by KamilDuljas** [comment]
H7 repo is also affected.

**Comment by KRASTM** [comment]
Hello @KamilDuljas,

Thank you for the report. 

Regarding the issue, and based on the RM either for F7 or H7:

- For the **TIM_CCER_CCxNE_MASK**: there are no CC4NE, CC5NE and CC6NE, so I think it's ok for this macro.

- For the  **TIM_CCER_CCxE_MASK:** I reported this one to our team in order to analyze it.

With regards,

**Comment by KRASTM** [comment]
ST Internal Reference: 195184

**Comment by TOUNSTM** [comment]
Fixed in ea67690c22fcf4683e8a1cda948b43cd0c77aa11
